boards/ti/lp_mspm0g3507/lp_mspm0g3507.yaml

```diff
@@ -0,0 +1,15 @@
+identifier: lp_mspm0g3507
+name: LP_MSPM0G3507
+type: mcu
```

Comment:
for the .dts, .yaml and _defconfig file please rename to `lp_mspm0g3507_mspm0g3507` then .dts, .yaml or _defconfig

Reply:
I don't understand the rationale behind this rename.

Reply:
As comment above